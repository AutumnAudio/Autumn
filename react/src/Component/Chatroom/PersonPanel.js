import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';
import PersonList from './PersonList';
import { leaveRoom } from '../AutumnApi/AutumnApi'
import {
    useHistory,
  } from "react-router-dom"
const styles = {
    roomHeader: {
        color: 'white',
        margin: '35px auto 0px 160px',
        display: 'inline-block',
    },
    arrowBack: {
        color: 'white',
        marginLeft: '25px',
    }
}
const leaveChatroom = (genre, history) => {
    console.log(genre)
    leaveRoom(genre).then((response) => {
        return response.text()
    }).then((text) => {
        console.log(text)
        history.push('/lobby')
    })
        
}
const PersonPanel = (props) => {
    const { classes, participants, genre } = props
    const history = useHistory()
    return (
        <div>
            <ArrowBackIcon className={classes.arrowBack} onClick={() => leaveChatroom(genre, history)}/>
            <h1 className={classes.roomHeader}>{genre}</h1>
            <PersonList participants={participants}/>
        </div>
    )
    
}

PersonPanel.propTypes = {
    classes: PropTypes.object.isRequired,
    participants: PropTypes.array.isRequired,
    genre: PropTypes.string.isRequired,
};

export default withStyles(styles)(PersonPanel)