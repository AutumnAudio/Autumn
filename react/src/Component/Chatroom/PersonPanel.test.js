import React from 'react'
import renderer from 'react-test-renderer'
import PersonPanel from './PersonPanel'

test('should be able to render person panel', () => {
    const participantsMock = [{username: 'testUser',
                        recentlyPlayed: [{name: 'song name', artists: ['testArtist']}]},
                        {username: 'testUser2',
                        recentlyPlayed: [{name: 'another song', artists: ['testArtist2']}]}]
    const genreMock = 'POP'
    const component = renderer.create(
        <PersonPanel participants={participantsMock} genre={genreMock}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
