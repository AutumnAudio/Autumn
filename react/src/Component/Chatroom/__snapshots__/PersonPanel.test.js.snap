// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be able to render person panel 1`] = `
<div>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root PersonPanel-arrowBack-2"
    focusable="false"
    onClick={[Function]}
    viewBox="0 0 24 24"
  >
    <path
      d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
    />
  </svg>
  <h1
    className="PersonPanel-roomHeader-1"
  >
    POP
  </h1>
  <div
    className="PersonList-personListDiv-3"
  >
    <div
      className="Person-personDiv-7"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root Person-personIcon-4"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
      <div
        className="Person-nameDiv-5"
      >
        testUser
      </div>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root Person-addIcon-8"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
      <div
        className="Person-currentSongDiv-6"
      >
        song name
         - 
        testArtist
      </div>
    </div>
    <div
      className="Person-personDiv-7"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root Person-personIcon-4"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
      <div
        className="Person-nameDiv-5"
      >
        testUser2
      </div>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root Person-addIcon-8"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
      <div
        className="Person-currentSongDiv-6"
      >
        another song
         - 
        testArtist2
      </div>
    </div>
  </div>
</div>
`;
